### PR TITLE
[RNMobile] Improve block inserter accessibility

### LINKED ISSUE #915: Improve Screen Reader support for Block Picker.
- The buttons content can be more friendly. (i.e. "Paragraph block. Button")
- Needs to set the `button` role.
- Mention that the block picker is scrollable.
- Give a hint when the picker is presented (currently it says nothing).

The Bottom Sheet accessibility improvements are marked separately on #914

### PR DESCRIPTION
**Pull request by jd-alexander**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/915

## Solution
The `MenuItem` that gets rendered for each block within the Inserter was given the role of `button` and it's accessibility label was appended with `block` to make it clearer that the options in the menu are to add blocks. 

The `open` event of the Inserter's bottom sheet now announces it's state so the user knows that there's a view that is overlaying the rest of the content. The copy is as follows. 

* When the sheet opens, the app announces `Scrollable block menu opened. Select a block.`
* When the sheet closes, the app announces `Scrollable block menu closed.`

## Concerns 

The only concern I have with this functionality is how I had to implement it on iOS. Firstly, I had to put a `delay` so that the announcement wasn't triggered during the screen changes else the announcement wouldn't be made at all. Also since an announcement is being made it's interrupting VoiceOver's behavior when it's attempting to refocus on the first element on the screen. 

<kbd></kbd>

I was thinking that I could add an [event listener](https://reactnative.dev/docs/accessibilityinfo#addeventlistener) to the custom announcements created and then [tell VoiceOver to focus ](https://reactnative.dev/docs/accessibilityinfo#setaccessibilityfocus)on those elements once the announcements are done. I wanted to try this but I wasn't sure how to get a reference to the first `View` on the screen. 
## Testing

1. Enable either VoiceOver on iOS or TalkBack on Android.
2. Open and close the bottom sheet to hear the announcement. 
3. Open the bottom sheet and hover over different blocks to hear the "block, Button" addition to each option. Before, only the name of the block would have been announced. 

### Android 
<details>
<summary>Click to see screenshots</summary>
<kbd></kbd>

<kbd></kbd>

<kbd></kbd>

* [Sheet opening announcement video](https://drive.google.com/file/d/1UznVky0cEDfK6OikF0dc9rBPiKB4nII3/view?usp=sharing)
* [Sheet closing announcement video](https://drive.google.com/file/d/1V0kcJLityThM8Qaoy1HfV71jBdwx9CKG/view?usp=sharing)
* [Announcement when a block is selected ](https://drive.google.com/file/d/1V2YaZZlaUg4CvAamHCJeJt-esHSyuMYW/view?usp=sharing)
</details>

### iOS

<details>
<summary>Click to see screenshots</summary>

<kbd></kbd>
<kbd></kbd>
<kbd></kbd>
</details>

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jd-alexander** [comment]
@etoledom could you please add yourself as a reviewer to this PR. Thank you 🙇

**Comment by jd-alexander** [comment]
I'm not sure why these tests are failing. How can these be re-run?

**Comment by jd-alexander** [comment]
After doing some more research within the codebase and elsewhere I realized there's an example of using a ref to set the accessibility focus. This was done in the [Add Mobile Spacer component PR](https://github.com/WordPress/gutenberg/commit/f73ebe6b644df05ad6f9fa1f5b3036b7eff6b8ff#diff-292ff6c089a7c791965593232b65e7f7R103). So my idea is to track which block is [currently focused using a custom hook.
](https://stackoverflow.com/questions/58886782/how-to-find-focused-react-component-like-document-activeelement) Once the custom announcement is done, then the previously focused area will be refocused. This solution seems like a lot of work to me just to get this behavior working. Sharing nonetheless, in case this can lead to other ideas.

**Comment by guarani** [comment]
> I'm not sure why these tests are failing. How can these be re-run?

It might be a permissions thing, do you see this re-run button after clicking on details of the failing test?

Screen Shot 2020-09-24 at 11 05 27

**Comment by jd-alexander** [comment]
Thanks for checking @guarani I am not able to see the re-run button so I think because I haven't merged in three PRs as yet I am not a contributor so I probably don't have that permission. When the three PRs I have opened are merged I should be good.

**Comment by guarani** [comment]
@jd-alexander I re-triggered them now.

> I am not able to see the re-run button so I think because I haven't merged in three PRs as yet I am not a contributor so I probably don't have that permission.

Gotcha 👌

**Comment by jd-alexander** [comment]
> @jd-alexander I re-triggered them now.
> 
> 
> 
> > I am not able to see the re-run button so I think because I haven't merged in three PRs as yet I am not a contributor so I probably don't have that permission.
> 
> 
> 
> Gotcha 👌
> 
> 

My changes probably broke some tests so I'm gonna do an investigation.

**Comment by jd-alexander** [comment]
I will be fixing these conflicts soon to get this in the review queue 🙏

**Comment by jd-alexander** [comment]
@etoledom the CI builds have been giving a lot of issues today hence the failure in the builds. You can still give this a try and let me know your thoughts on it. Thank you.

File: packages/components/src/mobile/inserter-button/index.native.js [hunk 1]
```diff
@@ -62,7 +62,8 @@ class MenuItem extends Component {
 				style={ styles.touchableArea }
 				underlayColor="transparent"
 				activeOpacity={ 0.5 }
-				accessibilityLabel={ blockTitle }
+				accessibilityRole="button"
+				accessibilityLabel={ `${ item.title } block` }
```
Comment (ceyhun) [inline packages/components/src/mobile/inserter-button/index.native.js:66]: This change might be breaking the e2e tests

Comment (jd-alexander) [reply packages/components/src/mobile/inserter-button/index.native.js:66]: Indeed. Thanks @ceyhun

**Comment by jd-alexander** [comment]
So after doing some debugging, I am getting a weird failure in my environment. When I run the `gutenberg-editor-gallery.test` locally on master or this branch it fails to find the block when this line is run in the test 
https://github.com/WordPress/gutenberg/blob/7532a485b400f86638145b71f94f6f717e5add25/packages/react-native-editor/__device-tests__/gutenberg-editor-gallery.test.js#L37

When I dug into the `getBlockAtPosition()` function I realized that the failure was occurring here
https://github.com/WordPress/gutenberg/blob/8feb3916d6dc72f10b965d4bd92109e6152f7f2b/packages/react-native-editor/__device-tests__/pages/editor-page.js#L47

Looking at the function parameters there is a default `position` of `1` and when I looked at my emulator I saw that the Gallery block was definitely not in the first position as the default editor was being shown like so

<kbd></kbd>

So I am wondering if the way how my test is being run locally is incorrect in some way. I just looked at another PR and saw the video of the test result and realized an empty editor is being used when the test was run.

**Comment by jd-alexander** [comment]
Yay! so the test was indeed fixed with my change 🥳 so I just need to get my local environment fixed then. @etoledom you can finally do a review of this. Thanks for the help @ceyhun I am hoping I can get my local environment in sync with CI. Did any of you encounter these issues? When I run the test suite I get a lot of failures. I am going to do a re-run and share the logs in Slack tomorrow.

**Comment by jd-alexander** [comment]
All is well here @etoledom for another review. This and the `gb-mobile` PR are passing as all test related issues have been resolved. Thanks much for all the help and guidance thus far 👍

**Review by etoledom — APPROVED** [review]
Tested on iOS and Android and it's working great!
Thank you for this improvement @jd-alexander 🙏

**Pull request merged**